src/runtime_src/core/tools/common/XBUtilities.cpp

```diff
@@ -308,4 +307,40 @@ XBUtilities::parse_device_indices(std::vector<uint16_t> &device_indices, const s
   } 
 }
 
+void
+XBUtilities::collect_devices( const std::set<std::string> &_deviceBDFs,
+                              xrt_core::DeviceCollection &_deviceCollection)
+{
+  // -- If the collection is empty then do nothing
+  if (_deviceBDFs.empty())
+    return;
+
+  // -- Collect all of devices if the "all" option is used...anywhere in the collection
+  if (_deviceBDFs.find("all") != _deviceBDFs.end()) {
+    uint64_t total = 0;
+    try {
+      // If there are no devices in the server a runtime exception is thrown in  mgmt.cpp probe()
+      total = xrt_core::get_total_devices(false /*isUser*/).first;
+    } catch (...) { 
+      /* Do nothing */ 
+    }
+
+    // No devices found
+    if (total == 0)
+      return;
+
+    // Now collect the devices and add them to the collection

```

Comment:
Maybe this function should take a parameter "is_user"? Right now xbmgmt is using "parse_device_indcies", but when we deprecate that, we want to get a list of mgmtpf devices and not userpf devices.

Reply:
Changes made as requested.